### PR TITLE
Pass content-IDs of all `ContentKey` elements to `Check`s

### PR DESCRIPTION
**Pull request by snazy**
Introduces the new type `IdentifiedContentKey`, which contains the content-ID for each content-key element.
Enhances the `VersionStore` interface to return `IdentifiedContentKey` types in `KeyEntry` and `Diff` and `ContentResult`.

### CODE REVIEW

**Comment by snazy** [comment]
Had to rebase due to merge conflicts

File: servers/quarkus-server/src/test/java/org/projectnessie/server/TestAuthorizationRules.java [hunk 1]
```diff
@@ -271,16 +277,37 @@ void testCanCommitButNotUpdateOrDeleteEntity() throws BaseNessieClientServerExce
                 role, key.toPathString()));
   }
 
+  @Test
+  // test_user3 has all permissions on a Branch, but ot permissions to DELETE_ENTITY
```
Comment (adutra) [inline servers/quarkus-server/src/test/java/org/projectnessie/server/TestAuthorizationRules.java:281]:
```suggestion
  // test_user3 has all permissions on a Branch, but no permissions to UPDATE_ENTITY
```

File: versioned/spi/src/main/java/org/projectnessie/versioned/ContentResult.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+package org.projectnessie.versioned;
+
+import javax.annotation.Nullable;
+import org.immutables.value.Value;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.Documentation;
+import org.projectnessie.model.IdentifiedContentKey;
+
+@Value.Immutable
+public interface ContentResult {
```
Comment (adutra) [inline versioned/spi/src/main/java/org/projectnessie/versioned/ContentResult.java:25]: `extends Result`?

File: servers/services/src/main/java/org/projectnessie/services/authz/Check.java [hunk 3]
```diff
@@ -47,29 +48,35 @@ public interface Check {
...
   static Check check(
       CheckType type,
       @Nullable @jakarta.annotation.Nullable NamedRef ref,
-      @Nullable @jakarta.annotation.Nullable ContentKey key,
-      @Nullable @jakarta.annotation.Nullable String contentId) {
-    return check(type, ref, key, contentId, null);
-  }
+      @Nullable @jakarta.annotation.Nullable IdentifiedContentKey identifiedKey) {
+    if (identifiedKey != null) {
+      IdentifiedContentKey.IdentifiedElement element = identifiedKey.lastElement();
```
[the remark this answers is not in the document]
Comment (snazy) [reply servers/services/src/main/java/org/projectnessie/services/authz/Check.java:69]: We don't check anything here. It's the responsibility of the SPI implementation to do the "hierarchical" checks.

File: versioned/spi/src/main/java/org/projectnessie/versioned/VersionStore.java [hunk 4]
```diff
@@ -329,7 +338,7 @@ PaginationIterator<KeyEntry> getKeys(
    * @return A parallel list of values.
    * @throws ReferenceNotFoundException if {@code ref} is not present in the store
    */
-  Map<ContentKey, Content> getValues(Ref ref, Collection<ContentKey> keys)
+  Map<ContentKey, ContentResult> getValues(Ref ref, Collection<ContentKey> keys)
```
[the remark this answers is not in the document]
Comment (snazy) [reply versioned/spi/src/main/java/org/projectnessie/versioned/VersionStore.java:341]: Could, but would prefer to do so in a follow-up.

File: versioned/spi/src/main/java/org/projectnessie/versioned/ResultType.java [hunk 5]
```diff
@@ -21,5 +21,6 @@ public enum ResultType {
   REFERENCE_CREATED,
   COMMIT,
   MERGE,
-  TRANSPLANT
+  TRANSPLANT,
+  CONTENT_RESULT
```
Comment (adutra) [inline versioned/spi/src/main/java/org/projectnessie/versioned/ResultType.java:25]: I'd note that the other constants don't have the `_RESULT` suffix... but I'm OK to leave it as is.

**Review by adutra — APPROVED** [review]

File: api/model/src/main/java/org/projectnessie/model/IdentifiedContentKey.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  @JsonSerialize(as = ImmutableIdentifiedElement.class)
+  @JsonDeserialize(as = ImmutableIdentifiedElement.class)
+  interface IdentifiedElement {
+    @Value.Parameter(order = 1)
+    String element();
+
+    @Value.Parameter(order = 2)
+    @Nullable
+    @jakarta.annotation.Nullable
+    String contentId();
```
Comment (dimas-b) [inline api/model/src/main/java/org/projectnessie/model/IdentifiedContentKey.java:74]: nit: should we make a note that all these elements (except the last one) are supported to be namespaces?

Comment (snazy) [reply api/model/src/main/java/org/projectnessie/model/IdentifiedContentKey.java:74]: I hope that this is clear ;)
(But actually: don't wanna trigger CI again ;) )

**Pull request merged**